metaflow/datatools/s3.py

```diff
@@ -809,18 +1014,30 @@ def get_all(self, return_info=False):
 
     def put(self, key, obj, overwrite=True, content_type=None, metadata=None):
         """
-        Put an object to S3.
-        Args:
-            key:           (required) suffix for the object.
-            obj:           (required) a bytes, string, or a unicode object to
-                           be stored in S3.
-            overwrite:     (optional) overwrites the key with obj, if it exists
-            content_type:  (optional) string representing the MIME type of the
-                           object
-            metadata:      (optional) User metadata to store alongside the object
-        Returns:
-            an S3 URL corresponding to the object stored.
+        Upload a single object to S3.
+
+        Parameters
+        ----------
+        key : str or `S3PutObject`
+            Object path. It can be an S3 url or a path suffix.
```

Comment:
I would clarify that for here, only the `key` part of `S3PutObject` is used and the metadata or content_type is ignored (and instead passed directly. Or just ignore `S3PutObject`.